internal/api/v1/admin/contacts/contacts_api.go

```diff
@@ -0,0 +1,96 @@
+package contacts
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/bitcoin-sv/spv-wallet-go-client/commands"
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/api/v1/errutil"
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/api/v1/querybuilders"
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/api/v1/user/contacts"
+	"github.com/bitcoin-sv/spv-wallet-go-client/queries"
+	"github.com/bitcoin-sv/spv-wallet/models/response"
+	"github.com/go-resty/resty/v2"
+)
+
+const route = "api/v1/admin/contacts"
+
+type API struct {
+	httpClient *resty.Client
+	url        *url.URL
+}
+
+func (a *API) Contacts(ctx context.Context, opts ...queries.ContactQueryOption) (*queries.UserContactsPage, error) {
+	var query queries.ContactQuery
+	for _, o := range opts {
+		o(&query)
+	}
+
+	queryBuilder := querybuilders.NewQueryBuilder(
+		querybuilders.WithMetadataFilter(query.Metadata),
+		querybuilders.WithPageFilter(query.PageFilter),
+		querybuilders.WithFilterQueryBuilder(&contacts.ContactFilterQueryBuilder{
+			ContactFilter: query.ContactFilter,
+			ModelFilterBuilder: querybuilders.ModelFilterBuilder{
+				ModelFilter: query.ContactFilter.ModelFilter,
+			},
+		}),
+	)
+	params, err := queryBuilder.Build()
+	if err != nil {
+		return nil, fmt.Errorf("failed to build admin contacts query params: %w", err)
+	}
+
+	var result queries.UserContactsPage
+	_, err = a.httpClient.
+		R().
+		SetContext(ctx).
+		SetResult(&result).
+		SetQueryParams(params.ParseToMap()).
+		Get(a.url.String())
+	if err != nil {
+		return nil, fmt.Errorf("HTTP response failure: %w", err)
+	}
+
+	return &result, nil
+}
+
+func (a *API) UpdateContact(ctx context.Context, cmd *commands.UpdateContact) (*response.Contact, error) {
+	var result response.Contact
+	_, err := a.httpClient.
+		R().
+		SetContext(ctx).
+		SetResult(&result).
+		SetBody(cmd).
+		Put(a.url.JoinPath(cmd.ID).String())
+	if err != nil {
+		return nil, fmt.Errorf("HTTP response failure: %w", err)
+	}
+
+	return &result, nil
+}
+
+func (a *API) RemoveContact(ctx context.Context, ID string) error {
```

Comment:
### very minor
Maybe `Delete` would be a better name for that as it is less error prone (Reject and Remove starts the same, and when using some IDE it could be easily selected by mistake)